Library/Homebrew/cmd/uninstall.rb

```diff
@@ -61,6 +77,30 @@ def uninstall
     end
   end
 
+  def should_check_for_dependents?
+    # --ignore-dependencies, to be consistent with install
+    return false if ARGV.include?("--ignore-dependencies")
+    return false if ARGV.homebrew_developer?
+    true
+  end
+
+  def check_for_dependents(kegs)
+    return false unless result = Keg.find_some_installed_dependents(kegs)
+
+    requireds, dependents = result
+
+    msg = "Refusing to uninstall #{requireds.join(", ")} because "
+    msg << (requireds.count == 1 ? "it is" : "they are")
+    msg << " required by #{dependents.join(", ")}, which "
+    msg << (dependents.count == 1 ? "is" : "are")
+    msg << " currently installed."
+    ofail msg
```

Comment:
It can be in another PR but I reckon a non-fatal version of this message should be printed in the developer case as a "Warning" rather than "Error".